### PR TITLE
fix(step-functions): fail gracefully on unsupported APIs

### LINKED ISSUE #454: Step Functions: AWS SDK Integration should fail with friendly message for invalid service calls
For all unsupported and deprecated calls, fail gracefully.

https://docs.aws.amazon.com/step-functions/latest/dg/connect-ddb.html

### PR DESCRIPTION
**Pull request by thantos**
Fixes #454 
Partial #443

Friendly error when unsupported SFN integration APIs are used.

https://docs.aws.amazon.com/step-functions/latest/dg/supported-services-awssdk.html

### CODE REVIEW

File: src/aws-sdk/sdk.ts [hunk 1]
```diff
@@ -58,6 +58,55 @@ class ServiceProxy {
   }
 }
 
+/**
+ * Step Functions AWS Service Integrations support integrations with many services, however, it has a list of
+ * services or methods that are not supported.
+ *
+ * https://docs.aws.amazon.com/step-functions/latest/dg/supported-services-awssdk.html
+ */
```
Comment (sam-goodwin) [inline src/aws-sdk/sdk.ts:66]: What happens when they add support for one? Are we then blocking users?

File: test/step-function.test.ts [hunk 2]
```diff
@@ -1519,6 +1519,37 @@ test("return AWS.SDK.CloudWatch.describeAlarms dynamic parameters", () => {
...
+          {
+            ConnectionId: "blah",
+            Data: "some data",
+          },
+          {
+            iam: {
+              resources: ["*"],
+            },
+          }
+        );
```
Comment (sam-goodwin) [inline test/step-function.test.ts:1540]: I thought we were going to polyfill this? Or are we planning to implement a different API call?

File: src/aws-sdk/sdk.ts (hunk 1 above)
Comment (thantos) [reply src/aws-sdk/sdk.ts:66]: Yeah, we could add an option to not validate.

Comment (sam-goodwin) [reply src/aws-sdk/sdk.ts:66]: Maybe these should be warnings, not blocking errors?

File: test/step-function.test.ts (hunk 2 above)
Comment (thantos) [reply test/step-function.test.ts:1540]: I was starting to look at implementing `API.Invoke`. We can add a convenience method for connections along with it, maybe.

`API.postConnection`
`API.getConnection`
`API.deleteConnection`

File: src/aws-sdk/sdk.ts [hunk 3]
```diff
@@ -58,6 +58,55 @@ class ServiceProxy {
...
+  EFS: ["createTags"],
+  ElasticTranscoder: ["testRole"],
+  EMR: ["describeJobFlows"],
+  Iot: [
+    "attachPrincipalPolicy",
+    "listPrincipalPolicies",
+    "detachPrincipalPolicy",
+    "listPolicyPrincipals",
+  ],
+  Lambda: ["invokeAsync"],
```
Comment (sam-goodwin) [inline src/aws-sdk/sdk.ts:102]: Oh no, this isn't supported? That seems ... so arbitrary ...

Comment (thantos) [reply src/aws-sdk/sdk.ts:102]: Its supported via the optimized integrations. Request/Response on a lambda.

> It is also possible to call Lambda asynchronously using the InvocationType parameter, as seen in the following example:

Comment (sam-goodwin) [reply src/aws-sdk/sdk.ts:102]: Is it possible to polyfill or is the interface botched?

File: test/step-function.test.ts (hunk 2 above)
Comment (thantos) [reply test/step-function.test.ts:1540]: To track.
https://github.com/functionless/functionless/issues/457

**Review by sam-goodwin — APPROVED** [review]

**Pull request merged**